Octokit/Http/SimpleJsonSerializer.cs

```diff
@@ -122,8 +122,19 @@ public override object DeserializeObject(object value, Type type)
                             //dictionary does not contain enum value and has no custom attribute. So add it for future loops and return value
                             // remove '-' from values coming in to be able to enum utf-8
                             stringValue = RemoveHyphenAndUnderscore(stringValue);
-                            var parsed = Enum.Parse(type, stringValue, ignoreCase: true);
-                            _cachedEnums[type].Add(value, parsed);
+                            object parsed = new object();
+                            try
+                            {
+                                parsed = Enum.Parse(type, stringValue, ignoreCase: true);
```

Comment:
We should use [`Enum.TryParse()`] (https://msdn.microsoft.com/en-us/library/dd783499(v=vs.110).aspx)  method here, rather than deliberately throwing/catching an exception